ta/pkcs11/src/processing_symm.c

```diff
@@ -149,34 +222,171 @@ allocate_tee_operation(struct pkcs11_session *session,
 	return tee2pkcs_error(res);
 }
 
+static enum pkcs11_rc pkcs2tee_load_hashed_attr(TEE_Attribute *tee_ref,
+						uint32_t tee_id,
+						struct pkcs11_object *obj,
+						enum pkcs11_attr_id pkcs11_id,
+						uint32_t tee_algo,
+						void *hash_ptr,
+						uint32_t *hash_size)
+{
+	TEE_OperationHandle handle = TEE_HANDLE_NULL;
+	void *a_ptr = NULL;
+	uint32_t a_size = 0;
+	TEE_Result res = TEE_ERROR_GENERIC;
+
+	if (get_attribute_ptr(obj->attributes, pkcs11_id, &a_ptr, &a_size))
+		return PKCS11_CKR_FUNCTION_FAILED;
+
+	/* Calculate hash of the attribute */
+	res = TEE_AllocateOperation(&handle, tee_algo, TEE_MODE_DIGEST, 0);
+	if (res) {
+		EMSG("TEE_AllocateOp. failed %#"PRIx32"", tee_algo);
+		return PKCS11_CKR_FUNCTION_FAILED;
+	}
+
+	res = TEE_DigestDoFinal(handle, a_ptr, a_size, hash_ptr, hash_size);
+	TEE_FreeOperation(handle);
+	if (res) {
+		EMSG("TEE_DigestOp. failed %#"PRIx32"", tee_algo);
```

Comment:
Can remove `""` after `PRIx32`
Also prefer replacing `"TEE_DigestOp. failed ..."` with explicit `"TEE_DigestDoFinal() failed ...` (or a generic `Digest failed ...`) as `TEE_DigestOp` does not match any valid label.

Ditto for line 244.

Reply:
Fixed